File: backend/cs/SocialNetApp/Bff/API/Dtos/NewUserDto.cs
using System.ComponentModel.DataAnnotations;

namespace Bff.API.Dtos
{
    public readonly record  struct NewUserDto
    {
        [Required]
        public string Name { get; init; }
        [Required]
        public string Surname { get; init; }
        [Required]
        public string Password { get; init; }
        public byte Age { get; init; }
        public string Sex { get; init; }
        public string City { get; init; }
        public string Info { get; init; }
    }
}

File: backend/go/socialnetapp/bff/internal/handler/dto/newuserdto.go
package dto

// NewUserDto represents dto for a newly registered user
type NewUserDto struct {
	Name     string  `json:"name"`
	Surname  string  `json:"surname"`
	Password string  `json:"password"`
	Age      *uint8  `json:"age"`
	Sex      *string `json:"sex"`
	City     *string `json:"city"`
	Info     *string `json:"info"`
}
